docs/cloud-docker-images.md

```diff
@@ -0,0 +1,100 @@
+# Sematic and Docker Images
+
+When Sematic runs your code in the cloud, it does so in a Docker container.
+Your code and all its dependencies need to be in that Docker image for it
+to run. Where does this Docker image come from? There are a few options, but
+a general theme is that Sematic prefers to make the construction of the Docker
+image transparent to you for simple cases, hooking into existing build tooling
+when possible. However, we leave the flexibility to customize for advanced
+usages if needed.
+
+## Image Construction
+### Bazel
+If you're using bazel, having Sematic create a cloud image containing your code
+is quite straightforward: we have a bazel macro that will allow you to create
+targets for building and pushing your image at the same time as creating a target
+to run the code locally. A complete example of this hookup can be found in our
+[bazel example repo](https://github.com/sematic-ai/example_bazel), but here's a
+summary of the steps. This assumes you already have a python bazel target to run
+a Sematic pipeline (we'll refer to the python script for this target your
+"launch script").
+
+1. Include Sematic's GitHub repo as a bazel repository in your bazel WORKSPACE
+2. Load Sematic's base images in your WORKSPACE, using, for example
+`load("@rules_sematic//:pipeline.bzl", "base_images")`, followed by
+`base_images()`
+3. Ensure you have a container registry where you can push your Docker images
+to
+4. In the bazel `BUILD` file where you have defined your launch script, load
+Sematic's pipeline macro:
+`load("@rules_sematic//:pipeline.bzl", "sematic_pipeline")`
+5. Replace the python binary target for your launch script with
+`sematic_pipeline`, using the same `deps` as you use for the binary target
+6. Fill out the `registry` and `repository` fields of the `sematic_pipeline`
+with information about where to push your image.
+
+With that, you're done! Assuming the target for your launch script was
+`//my_repo/my_package:my_target`, you now have the following targets available:
+
+- `//my_repo/my_package:my_target`: still runs your target, but builds a Docker
+image with your code and its dependencies first
+- `//my_repo/my_package:my_target_local`: runs your target WITHOUT building and
+pushing the image. This can help for local development when you don't want the
+overhead of waiting for the build & push.
+- `//my_repo/my_package:my_target_image`: builds the image, but doesn't push it
+or run your script
+- `//my_repo/my_package:my_target_push`: builds and pushes the image, but
+doesn't run your script
+
+#### Custom base images
+The `sematic_pipeline` macro also allows you to specify a custom base image to
+cover any dependencies you have that aren't specified in bazel. You can do this
+by setting the `base` field of the `sematic_pipeline` macro, If, for
```

Comment:
Maybe say:
"""
You can do this as follows:
```
sematic_pipeline(
    name = "my_pipeline",
    base = "uri://to/my/base/image",
    ...
)
```
"""

Reply:
done